pkg/storaged/client.js

```diff
@@ -769,6 +790,94 @@
 
     client.vdo_overlay = vdo_overlay();
 
+    /* HACK - clevis */
+
+    function clevis_overlay() {
+        var self = {
+            start: start,
+
+            info: { },
+
+            find_by_block: find_by_block,
+
+            remove: remove,
+            get_adv: get_adv,
+            add: add,
+            check_key: check_key,
+            replace: replace,
+            unlock: unlock
+        };
+
+        function spawn_tool(args) {
+            return cockpit.spawn([ "python3", "--", "-" ].concat(args), { superuser: "try", err: "message" })
+                .input(cockpit_clevis_tool_py);
+        }
+
+        function start() {
+            var buf = "";
+            spawn_tool([ "monitor" ])
+                .stream(function (output) {
+                    var lines;
+
+                    buf += output;
+                    lines = buf.split("\n");
+                    buf = lines[lines.length-1];
+                    if (lines.length >= 2) {
+                        self.info = JSON.parse(lines[lines.length-2]);
+                        client.dispatchEvent("changed");
+                    }
+                }).
+                fail(function (error) {
+                    if (error != "closed") {
+                        console.warn(error);
+                    }
+                });
+        }
+
+        function some(array, func) {
+            var i;
+            for (i = 0; i < array.length; i++) {
+                var val = func(array[i]);
+                if (val)
+                    return val;
+            }
+            return null;
+        }
```

Comment:
This looks like `array.find(val => func(val))`?

Reply:
You mean `array.find(func)`? :-)  `some` returns the first non-false return value of `func`, while `find` returns a element of the array itself.